#! /usr/bin/env python
# DTlanner configuration

PACKAGE='dt_local_planner'

import sys

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("max_vel", double_t, 0,
    "The maximum velocity for the robot in m/s", 0.30)

gen.add("max_ang", double_t, 0,
    "The maximum angular speed for the robot in 1/s", 0.30)

gen.add("max_vel_deltaT0", double_t, 0,
    "The maximum velocity between each points of the spline in m/s", 0.30)

gen.add("K_p", double_t, 0,
    "gain control param for the dynamic trajectory controller", 1.0)

gen.add("K_d", double_t, 0,
    "differential control param for the dynamic trajectory controller", 1.0)

gen.add("stepCnt_max_vel", int_t, 0,
    "The amount of timesteps until the maximum velocity applys.", 10, 1)

gen.add("nPolyGrad", int_t, 0,
    "The grade of the polynomial, which gets generated from polyfit", 15, 1)

gen.add("move", bool_t, 0,
    "Enable publishing of movement commands", True)

exit(gen.generate(PACKAGE, "dt_local_planner", "DTPlanner"))
